src/exercise.rs

```diff
@@ -47,9 +47,34 @@ pub struct ContextLine {
     pub important: bool,
 }
 
+pub struct CompiledExercise<'a> {
+    exercise: &'a Exercise,
+    _handle: FileHandle,
+}
+
+impl<'a> CompiledExercise<'a> {
+    pub fn run(&self) -> Result<ExerciseOutput, ExerciseOutput> {
+        self.exercise.run()
+    }
+}
+
+struct FileHandle;
+
+#[derive(Debug)]
+pub struct ExerciseOutput {
+    pub stdout: String,
+    pub stderr: String,
+}
+
+impl Drop for FileHandle {
```

Comment:
This is a silly trick to avoid calling `clean()` all over the place. I think it's not racier than what we were doing before.

Reply:
:+1: